snow/consensus/snowball/snowflake_plus.go

```diff
@@ -0,0 +1,116 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package snowball
+
+import (
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/bag"
+)
+
+var _ Consensus = (*snowflakePlus)(nil)
+
+// snowflakePlus implements Consensus using Snowflake with multiple
+// alpha thresholds and corresponding beta values.
+type snowflakePlus struct {
+	// alpha1 is the threshold of votes required to update the preference
+	// of the consensus instance.
+	alpha1 int
+
+	// alpha2Cutoff is the lowest alpha threshold that can be used to find
+	// the termination point.
+	alpha2Cutoff int
+
+	// betas are the number of required successful polls corresponding to
+	// each alpha threshold required to finalize.
+	// The ith element of betas is the number of consecutive polls that
+	// received >= alpha2Cutoff + i votes required to finalize.
+	betas []int
+
+	// confidence tracks the number of successful polls in a row for each
+	// corresponding alpha threshold.
+	// The ith element of confidence is the number of consecutive polls
+	// that received >= alpha2Cutoff + i votes.
+	confidence []int
+
+	finalized bool
+
+	choice ids.ID
```

Comment:
`choice` makes it sound like this is final... I feel like `preference` (as used in the other implementations) is more clear.